### PR TITLE
Fix modding history not applying user filter

### LINKED ISSUE #3145: User Modding History is broken and showing ..collective history?
If you go to any users modding history page, you will currently see the same thing on every page: what appears to be activity for ALL users. 

The only sections currently working correctly and being different depending on which user's page you're seeing is "Most upvoted by" and "Most upvoted"

### PR DESCRIPTION
**Pull request by notbakaneko**
fixes #3145 

`request()->all()`/`request()->toArray()` also does not have the same values as `request()`

### CODE REVIEW

**Comment by nanaya** [comment]
wtf is this "checks" tab ヽ( ﾟ ヮﾟ)ﾉ

File: app/Models/BeatmapDiscussion.php [hunk 1]
```diff
@@ -105,7 +105,7 @@ public static function search($rawParams = [])
             $query->withoutDeleted();
         }
 
-        if (!($params['is_moderator'] ?? false)) {
+        if (!($rawParams['is_moderator'] ?? false)) {
```
Comment (nanaya) [inline app/Models/BeatmapDiscussion.php]: ( ¬‿¬)

Comment (notbakaneko) [reply app/Models/BeatmapDiscussion.php]: good thing defaults to `false` ( ¬‿¬)

Comment (peppy) [reply app/Models/BeatmapDiscussion.php]: can we stick to using sentences to comment on code rather than faces? it's very hard to follow along.
